library/proc_macro/src/bridge/buffer.rs

```diff
@@ -78,18 +78,40 @@ impl<T: Copy> Buffer<T> {
         mem::take(self)
     }
 
+    pub(super) fn extend_from_array<const N: usize>(&mut self, xs: &[T; N]) {
```

Comment:
Could you add a comment telling why having `N` to be a constant is important?
Otherwise it's easy to remove this function as redundant.